### DISCUSSION TITLE
Fluent mapping behavior

### DISCUSSION THREAD

**Discussion by smalleythedave** [linq2db/linq2db#3118] [General]
I encountered something I don't understand when using fluent mapping. If I have an entity with a property like "Id" and apply an attribute like thus:

`[Column("USER_ID", IsIdentity = true, IsPrimaryKey = true)]`
`public virtual long UserId { get; set; }`

everything works as expected. However, if I use a fluent mapping:

`builder.Entity<User>()`
`    .HasTableName("USERS")`
`    .Property(u => u.UserId).HasAttribute(new ColumnAttribute("USER_ID) { IsIdentity = true, IsPrimaryKey = true });`

it appears the mapping isn't active. I get errors about how "UserId" is not a column or there is no identity defined. If, however, I do the fluent mapping like this:

`builder.Entity<User>()`
`    .HasTableName("USERS")`
`    .Property(u => u.UserId).HasColumnName("USER_ID").IsIdentity().IsPrimaryKey();`

everything works like the attributed version.

So is this expected behavior? Why can't I "apply the attribute" with the fluent builder? I believe this is the way associations are supposed to be defined fluently (with "HasAttribute()"). I can live with it, but it seems counter-intuitive.

Dave

**Comment by MaceWindu** [comment]
When you call `Property`, fluent mapping builder already creates `ColumnAttribute` internally and then you add anoter one, but we don't merge it's properties into existing attribute and instead add another one which is ignored as we take only first instance of aattribute into account. I've created issue #3119 to track it
